tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+import random
+
+import numpy as np
+import pytest
+import tensorflow as tf
+from common.tf_layer_test_class import CommonTFLayerTest
+
+rng = np.random.default_rng(475912)
+
+
+class TestSparseSegmentMean(CommonTFLayerTest):
+    def _prepare_input(self, inputs_info):
+        assert 'indices:0' in inputs_info
+        assert 'values:0' in inputs_info
+        assert 'segment_indices:0' in inputs_info
+
+        values_shape = inputs_info['values:0']
+
+        inputs_data = {}
+        inputs_data['values:0'] = rng.uniform(-5.0, 5.0, values_shape).astype(self.data_type)
+
+        # generate all possible indices
+        all_indices = []
+        for row_ind in range(0, self.shape):
+            all_indices.append(row_ind)
+        inputs_data['indices:0'] = rng.choice(all_indices, self.shape, replace=False).astype(self.indices_type)
+
+        segment_ids = []
+        for ind in range(0, self.shape):
+            segment_ids.append(self.indices_type(random.randint(0, self.segments_num)))
+        inputs_data['segment_indices:0'] = sorted(segment_ids)
+
+        return inputs_data
+
+    def create_sparse_segment_mean(self, data_type, indices_type,
+                                  shape, segments_num):
+        self.data_type = data_type
+        self.indices_type = indices_type
+        self.shape = shape
+        self.segments_num = segments_num
+        tf.compat.v1.reset_default_graph()
+        with tf.compat.v1.Session() as sess:
+            values = tf.compat.v1.placeholder(data_type, [shape], 'values')
+            indices = tf.compat.v1.placeholder(indices_type, [shape], 'indices')
+            segments_ids = tf.compat.v1.placeholder(indices_type, [shape], 'segment_indices')
+            tf.raw_ops.SparseSegmentMean(
+                data=values,
+                indices=indices,
+                segment_ids=segments_ids)
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        return tf_net, None
+
+    @pytest.mark.parametrize('data_type', [np.float32, np.float64, np.float16])
+    @pytest.mark.parametrize('indices_type', [np.int32, np.int64])
```

Comment:
please have parameter type for `indices` and `segment_ids` separately. They may have different types, for example. one - of int32 and other - of int64

Reply:
Done.